packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -0,0 +1,729 @@
+    );
+  }
+
+  /// Linearly interpolate between two slider themes.
+  ///
+  /// The arguments must not be null.
+  ///
+  /// The `t` argument represents position on the timeline, with 0.0 meaning
+  /// that the interpolation has not started, returning `a` (or something
+  /// equivalent to `a`), 1.0 meaning that the interpolation has finished,
+  /// returning `b` (or something equivalent to `b`), and values in between
+  /// meaning that the interpolation is at the relevant point on the timeline
+  /// between `a` and `b`. The interpolation can be extrapolated beyond 0.0 and
+  /// 1.0, so negative values and values greater than 1.0 are valid (and can
+  /// easily be generated by curves such as [Curves.elasticInOut]).
+  ///
+  /// Values for `t` are usually obtained from an [Animation<double>], such as
+  /// an [AnimationController].
+  static SliderThemeData lerp(SliderThemeData a, SliderThemeData b, double t) {
+    assert(a != null);
+    assert(b != null);
+    assert(t != null);
+    return new SliderThemeData(
+      activeRailColor: Color.lerp(a.activeRailColor, b.activeRailColor, t),
+      inactiveRailColor: Color.lerp(a.inactiveRailColor, b.inactiveRailColor, t),
+      disabledActiveRailColor: Color.lerp(a.disabledActiveRailColor, b.disabledActiveRailColor, t),
+      disabledInactiveRailColor:
+          Color.lerp(a.disabledInactiveRailColor, b.disabledInactiveRailColor, t),
+      activeTickMarkColor: Color.lerp(a.activeTickMarkColor, b.activeTickMarkColor, t),
+      inactiveTickMarkColor: Color.lerp(a.inactiveTickMarkColor, b.inactiveTickMarkColor, t),
+      disabledActiveTickMarkColor:
+          Color.lerp(a.disabledActiveTickMarkColor, b.disabledActiveTickMarkColor, t),
+      disabledInactiveTickMarkColor:
+          Color.lerp(a.disabledInactiveTickMarkColor, b.disabledInactiveTickMarkColor, t),
+      thumbColor: Color.lerp(a.thumbColor, b.thumbColor, t),
+      disabledThumbColor: Color.lerp(a.disabledThumbColor, b.disabledThumbColor, t),
+      overlayColor: Color.lerp(a.overlayColor, b.overlayColor, t),
+      valueIndicatorColor: Color.lerp(a.valueIndicatorColor, b.valueIndicatorColor, t),
+      thumbShape: t < 0.5 ? a.thumbShape : b.thumbShape,
+      valueIndicatorShape: t < 0.5 ? a.valueIndicatorShape : b.valueIndicatorShape,
+      showValueIndicator: t < 0.5 ? a.showValueIndicator : b.showValueIndicator,
+    );
+  }
+
+  @override
+  int get hashCode {
+    return hashValues(
+      activeRailColor,
+      inactiveRailColor,
+      disabledActiveRailColor,
+      disabledInactiveRailColor,
+      activeTickMarkColor,
+      inactiveTickMarkColor,
+      disabledActiveTickMarkColor,
+      disabledInactiveTickMarkColor,
+      thumbColor,
+      disabledThumbColor,
+      overlayColor,
+      valueIndicatorColor,
+      thumbShape,
+      valueIndicatorShape,
+      showValueIndicator,
+    );
+  }
+
+  @override
+  bool operator ==(Object other) {
+    if (other.runtimeType != runtimeType) {
```

Comment:
you might want to do an `identical()` test here first since there's so many subproperties and they're so often going to be identical.

Reply:
OK, done.